app/controllers/api/v2/hosts_controller.rb

```diff
@@ -279,7 +279,7 @@ def power_status
           statusText: _("Failed to fetch power status: %s") % e,
         }
 
-        render json: resp.merge(PowerManager::PowerStatus::HOST_POWER[:na])
+        render json: resp.merge(PowerManager::PowerStatus::HOST_POWER[:na]), status: :internal_server_error
```

Comment:
Could we use a 4xx code instead of 500? 5xx codes are usually hard failures, but we've recover from it. And 500 is usually 'oh boy, we have no idea what could go wrong'. From the [Rails implemented error codes](http://billpatrianakos.me/blog/2013/10/13/list-of-rails-status-code-symbols/) I would suggest using `:unprocessable_entity` or `:failed_dependency`.
*If you want go rogue you could use unofficial 420 - I just love that one* :eyes: :smile:

Reply:
Thank you for the list !
from what I understand, the 4xx codes are for client errors, e.g:
![Selection_237](https://user-images.githubusercontent.com/26363699/69696706-33c20880-10e9-11ea-82c1-7d15ab3b67cc.png)
 it sais that the 4xx class of status code is intended for situations in which the error seems to have been caused by the client. 

I guess in this case it is a server error, the `statusText` that we are returning from the server on failure is: `Failed to fetch power status`

So which status would we choose from the given list?: 
![Selection_238](https://user-images.githubusercontent.com/26363699/69696866-a7641580-10e9-11ea-884b-ff469f011752.png)

I couldn't find something that really suits this case other than the generic `500`,
for more info about the 500 codes: https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#5xx_Server_errors

imho it would be fine to keep it `500` with the `statusText` that explains exactley what went wrong. eventually the user will see that message and not the status code.

Reply:
Ok, make sense. :+1: It's not really error on server, but I'm ok with this :)

Reply:
Overall I'd say 400 is certainly incorrect since the user did nothing wrong. However, I also kind of disagree with 500. In many applications you monitor for HTTP 500 to find application issues. It's a bit of a gray area to say whether this is an application error. The system may be offline and nothing Foreman can do about this. The argument can be made that the response should be that the state is `unknown` and it should be a regular API response. That was the previous behavior and I'd say that's actually correct.

A general rule of thumb I keep is that an application server should never generate a HTTP 500. Anything that's 500 should be treated as a bug.

Reply:
From my understanding the previous behavior returned status code `200` which translated as success in the UI instead of failure as it should. and I think `500` is just a general error isn't it? I definitely agree it's a gray area :rofl:

Reply:
500 is usually an unexpected error that wasn't handled properly. In this case we have error handling. The UI may need to have special handling for the unknown state. However, in case of the host table it's probably sufficient to show an icon for unknown. With this you can't just map `na` to `na.png` (or whatever) and need special handling. I'd actually consider reverting this.

Reply:
I wouldn't suggest to revert this as it was returning a "success" earlier,
but if you come up to a better error status code I would go with it :+1:

Reply:
As an API consumer I'm used to writing code that raises an exception on HTTP 500. In this case I can handle `unknown` just fine I think, but can you show me where you actually care whether it's `unknown` or actually on/off?

Reply:
in the [UI redux](https://github.com/theforeman/foreman/pull/7210/files?file-filters%5B%5D=.js#diff-75a58a82eba61d46d18413db986176b4R17) we don't care for a spcific error though we need the API call to treat it as an error returned in the `catch` [part](https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/redux/API/APIRequest.js#L18).

Discussing it offline with @ezr-ondrej we agree to use the 424 - Failed Dependency:
"The request failed because it depended on another request and that request failed"
how does it sound?

Reply:
I don't understand why you need to treat it as an error in the first place. Can you show me a screenshot how it's presented in the UI?

Reply:
I agree, I was too fast to merge. 500 could mess with some clients, @laviro will you open another PR to fix this, or should we revert?

Reply:
I wanted to discuss first to which status would we change? Also opened a dicussion about it on discourse, @ekohl luckily in this case the visibility doesn't change.. but the path that the data is moving is wrong as you get an error messages on a success result.

I also understand your point to keep the API as it was.. what if we used the 204 success code: "204 No Content: The server successfully processed the request and is not returning any content" . It won't change the API to work as before and maybe in the Redux we could listen to 204 codes and treat them as errors? cc @theforeman/ui-ux

Reply:
If you won't to revert, I wouldn't change the UI part of this PR as it wasn't correct earlier, but I don't want to block you from returning the API to what it was if it's urgent